src/routes/SandboxPage/SandboxPage.tsx

```diff
@@ -1,21 +1,52 @@
 import React from 'react';
-import { Alert } from '@patternfly/react-core/dist/dynamic/components/Alert';
-import { AlertVariant } from '@patternfly/react-core/dist/dynamic/components/Alert';
+import { Alert, AlertVariant } from '@patternfly/react-core/dist/dynamic/components/Alert';
 import { Bullseye } from '@patternfly/react-core/dist/dynamic/layouts/Bullseye';
 import { Flex } from '@patternfly/react-core/dist/dynamic/layouts/Flex';
 import { PageSection } from '@patternfly/react-core/dist/dynamic/components/Page';
 import { Spinner } from '@patternfly/react-core/dist/dynamic/components/Spinner';
-import { Text } from '@patternfly/react-core/dist/dynamic/components/Text';
-import { TextContent } from '@patternfly/react-core/dist/dynamic/components/Text';
-import { TextVariants } from '@patternfly/react-core/dist/dynamic/components/Text';
+import {
+  Text,
+  TextContent,
+  TextVariants,
+} from '@patternfly/react-core/dist/dynamic/components/Text';
 import SandboxPageBanner from '../../components/PageBanner/SandboxPageBanner';
 import HowItWorksCard from '../../components/HowItWorksCard/HowItWorksCard';
 import GetStartedCard from '../../components/GetStartedCard/GetStartedCard';
 import ServiceCatalog from '../../components/ServiceCatalog/ServiceCatalog';
 import { useRegistrationContext } from '../../hooks/useRegistrationContext';
+import useRegistrationService from '../../hooks/useRegistrationService';
+import { canaryDeploymentCheck } from '../../utils/canary-deployment';
 
 const SandboxPage = () => {
   const [{ status, error }] = useRegistrationContext();
+
+  /*** CANARY Deployment for redirecting users to new UI
+   Users visiting the first time the UI will get assigned with a random number, the number will be stored in localstorage and reused everytime the user opens the UI.
+   The users that gets a number that is lower than the canary weight threshold, they will be redirected to the new ui,
+   the others will be using the current UI.
+   The threshold can be configured in the backend so that a bigger/smaller number of users can be routed to the new UI.
+   ***/
+  const loadUIConfig = async function () {
+    console.log('loading ui config');
+    const { getUIConfigData } = useRegistrationService();
+    const uiConfigData = await getUIConfigData();
+    if (uiConfigData == undefined) {
+      console.log('Unable to load uiconfig. Got undefined response.');
+    } else {
+      console.log('uiCanaryDeploymentWeight found: ', uiConfigData.uiCanaryDeploymentWeight);
+      canaryDeploymentCheck(uiConfigData.uiCanaryDeploymentWeight);
+    }
+  };
+  /** END canary UI deployment logic **/
+
+  (async () => {
+    try {
+      await loadUIConfig();
```

Comment:
@mfrancisc won't this be called every re-render of the `SandboxPage`? Can we use a relevant react hook to accomplish this and prevent unnecessary calls during renders?

Reply:
this needs to be executed before loading the component . I mean we first need to see if the user should stay on this UI or not, and than return the component. Would work in the same way with the hook ?

Reply:
I've implemented the hook, and moved it to the Routing component, as that is more appropriate since it needs to redirect before displaying the sandbox page.